packages/bsky/src/api/app/bsky/feed/getAuthorFeed.ts

```diff
@@ -79,21 +79,47 @@ export const skeleton = async (inputs: {
   if (clearlyBadCursor(params.cursor)) {
     return { actor, filter: params.filter, items: [] }
   }
+
+  const isFirstPageRequest = !params.cursor
+  const shouldInsertPinnedPost =
+    isFirstPageRequest && params.includePins && !!actor.profile?.pinnedPost
+
   const res = await ctx.dataplane.getAuthorFeed({
     actorDid: did,
     limit: params.limit,
     cursor: params.cursor,
     feedType: FILTER_TO_FEED_TYPE[params.filter],
   })
+
+  let items = res.items.map((item) => ({
+    post: { uri: item.uri, cid: item.cid || undefined },
+    repost: item.repost
+      ? { uri: item.repost, cid: item.repostCid || undefined }
+      : undefined,
+  }))
+
+  if (shouldInsertPinnedPost && actor.profile?.pinnedPost) {
```

Comment:
Nice! You can lose the additional `actor.profile?.pinnedPost` too 👍

Reply:
nope I cannot - TypeScript gets mad :(